### PR TITLE
Validating image from URI

### PR DESCRIPTION
**Pull request by hugoseabra-imachines**
Validation whether a given URI in `datapoint_uri` has a valid image.

IMI-4164

### CODE REVIEW

File: basemodels/pydantic/utils.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+logger = logging.getLogger(__name__)
+
+
+class URIRequestError(UrlError):
+    """ Raises when URL is invalid.  """
+    pass
+
+
+def check_valid_image(name: str, uri: str) -> None:
```
Comment (igervasiychuk) [inline basemodels/pydantic/utils.py:15]: there seem to be two copy of this func?

File: basemodels/manifest/data/taskdata.py [hunk 2]
```diff
@@ -41,4 +43,7 @@ def validate_taskdata_entry(value: dict):
     if not isinstance(value, dict):
         raise ValidationError("taskdata entry should be dict")
 
-    TaskDataEntry(value).validate()
+    task_data = TaskDataEntry(value)
+    task_data.validate()
+
+    check_valid_image(str(task_data.task_key), str(task_data.datapoint_uri))
```
Comment (igervasiychuk) [inline basemodels/manifest/data/taskdata.py:49]: this would validate **every** image, we need just one

File: basemodels/manifest/utils.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    Checks whether a content from URI is a valid image.
+
+    Parameters:
+        name(str): name referenced for URI value
+        uri(str): URI where content must be fetched
+    Raises:
+        ValidationError: invalid image
+    """
+    try:
+        image_formats = ("image/png", "image/jpeg", "image/jpg")
```
Comment (igervasiychuk) [inline basemodels/manifest/utils.py:23]: this doesn't make sense here (just three values), but in general, it'd be nice for it to be a set

File: basemodels/manifest/utils.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        r = requests.head(uri)
+
+        if 'content-type' in r.headers and r.headers["content-type"] not in image_formats:
```
Comment (igervasiychuk) [inline basemodels/manifest/utils.py:26]: may be r.headers('content-type','') not in ....
also, inconsistent quote symbols (' and " in one line)

File: tests/test_utils.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+from basemodels.pydantic import TaskData
+from basemodels.manifest.data.taskdata import TaskDataEntry
+from basemodels.manifest.utils import (
+    check_valid_image as schematicsImageChecker,
+    URIRequestError as SchematicsRequestError
+)
+
+from basemodels.pydantic.utils import (
+    check_valid_image as pydanticImageChecker,
```
Comment (igervasiychuk) [inline tests/test_utils.py:14]: unclear: 
why rename it
why a function is camelCase

**Review by igervasiychuk — CHANGES_REQUESTED** [review]

File: basemodels/manifest/manifest.py [hunk 6]
```diff
@@ -175,8 +176,8 @@ class Manifest(Model):
     job_id = UUIDType(default=uuid.uuid4)
     job_total_tasks = IntType(required=True)
     network = StringType(required=False)
-    only_sign_results = BooleanType(default=False,)
-    public_results = BooleanType(default=False,)
+    only_sign_results = BooleanType(default=False, )
```
Comment (igervasiychuk) [inline basemodels/manifest/manifest.py:179]: can we remove ", "

**Review by igervasiychuk — CHANGES_REQUESTED** [review]

**Review by igervasiychuk — COMMENTED** [review]
pls test it in our env

**Comment by igervasiychuk** [comment]
+ Will that work with text challenges

**Comment by igervasiychuk** [comment]
also, will that work with taskdata, and groundtruth? (not just taskdata_uri, groundtruth_uri)

File: basemodels/manifest/manifest.py [hunk 7]
```diff
@@ -320,8 +321,28 @@ def validate_manifest_uris(manifest: dict):
             response = requests.get(uri)
             response.raise_for_status()
 
-            entries_count = traverse_json_entries(response.json(), validate_entry)
-        except (BaseError, RequestException) as e:
+            data = response.json()
```
Comment (e271828-) [inline basemodels/manifest/manifest.py:324]: This needs to be rethought. You're loading a potentially gigantic file into memory with no safeguards. Maybe cap it to first 100KB.

**Comment by igervasiychuk** [comment]
don't forget to bump the version

**Pull request closed without merging**